google/resource_storage_bucket.go

```diff
@@ -406,17 +406,24 @@ func resourceStorageBucketRead(d *schema.ResourceData, meta interface{}) error {
 		return handleNotFoundError(err, d, fmt.Sprintf("Storage Bucket %q", d.Get("name").(string)))
 	}
 
-	log.Printf("[DEBUG] Read bucket %v at location %v\n\n", res.Name, res.SelfLink)
 	// We need to get the project associated with this bucket because otherwise import
 	// won't work properly.  That means we need to call the projects.get API with the
 	// project number, to get the project ID - there's no project ID field in the
-	// resource response.
-	log.Println("[TRACE] Fetching project ID.")
-	proj, err := config.clientCompute.Projects.Get(strconv.FormatUint(res.ProjectNumber, 10)).Do()
-	if err != nil {
-		return err
+	// resource response.  However, this requires a call to the Compute API, which
+	// would otherwise not be required for this resource.  So, we're going to
+	// intentionally check whether the project is set *on the resource*.  If it is,
+	// we will not try to fetch the project name.  If it is not, either because
+	// the user intends to use the default provider project, or because the resource
+	// is currently being imported, we will read it from the API.
+	if p, ok := d.GetOk("project"); ok && p != "" {
```

Comment:
No need to check that p != "", that's what ok means.

Also shouldn't this be checking that project isn't set?

Reply:
Oof, the fact that that made the tests pass makes me nervous.  Adding a new test.